### ISSUE TITLE
Does YOLOV5 classification support the `resume training`

### ISSUE THREAD

**Issue by AI-Passionner** [ultralytics/yolov5#8728]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

See above. 
Thanks.

**Comment by pourmand1376** [comment]
Yes, use `--resume` parameter when training. Also, you can load your previous trained models using `--weight` parameter and use it to resume your training.

**Comment by AI-Passionner** [comment]
@pourmand1376 Thank you. 
That works for object detection. Classification might not support yet.

**Comment by pourmand1376** [comment]
Hmm. That I don't know! Thanks for further clarification. @AI-Passionner

**Comment by glenn-jocher** [comment]
TODO: Classification --resume

**Comment by RamatovInomjon** [comment]
> TODO: Classification --resume

Hi , Thanks for the great work! Classification resume is done?

**Comment by glenn-jocher** [comment]
@RamatovInomjon no, not started yet (extremely overloaded with tasks). If you'd like to contribute an implementation PR that would really help.

The fastest and easiest way to incorporate your ideas into the official codebase is to submit a Pull Request (PR) implementing your idea, and if applicable providing before and after profiling/inference/training results to help us understand the improvement your feature provides. This allows us to directly see the changes in the code and to understand how they affect workflows and performance.

Please see our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/) to get started.

**Comment by IyazGreat** [comment]
HI @glenn-jocher , I trust you are well. 

Any update on this? 

Thanks!

**Comment by glenn-jocher** [comment]
@IyazGreat Hi there! Unfortunately, the classification resume feature has not been implemented yet. We are currently busy with various tasks, but we are always open to community contributions. If you'd like to help with this feature, please consider submitting a Pull Request following our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Your input would be greatly appreciated.

**Comment by Blessing988** [comment]
I have trained my YOLOv5 classication model for some epochs and I want to resume it. The --`--resume` argument isn't in the training script same as the `weights` argument. How do I continue training the model?

**Comment by glenn-jocher** [comment]
@Blessing988 for resuming training of your YOLOv5 classification model, while the `--resume` argument is specifically designed for object detection tasks and might not be directly applicable for classification tasks in its current form, you can still effectively "resume" your training by using the `--weights` argument. Here's how you can do it:

1. When you initially start your training, YOLOv5 saves checkpoints of your model at regular intervals in the `runs/train/exp*/weights` directory (where `*` is the experiment number).
2. To resume training, you can use the last saved checkpoint by specifying it with the `--weights` parameter. For example, if your last checkpoint is `best.pt`, you would use it like this:

```bash
python train.py --img 640 --batch 16 --epochs 100 --data your_dataset.yaml --weights runs/train/exp*/weights/best.pt
```

This command tells YOLOv5 to start training using the weights from your last checkpoint, effectively resuming your training.

Remember, while this method allows you to continue training from where you left off, it's always a good idea to keep an eye on your training metrics to ensure everything is proceeding as expected.

**Comment by Blessing988** [comment]
@glenn-jocher Thank you very much.

**Comment by glenn-jocher** [comment]
@Blessing988 you're welcome! If you have any more questions or need further assistance as you continue with your project, feel free to reach out. Happy training! 🚀

**Comment by Blessing988** [comment]
@glenn-jocher Also for YOLOv5 classification model, does it have the same `backbone` and `neck` as the ones for object detection? If yes, how many conv blocks are in the head. 

Can you let me know the architecture overview of the classification models as I can't find the yaml file for it. I would be glad. Thanks.

**Comment by glenn-jocher** [comment]
@Blessing988, yes, YOLOv5 classification models share the same `backbone` and `neck` as the object detection models. The main difference lies in the `head` structure tailored for classification tasks. The specific number of convolutional blocks in the head may vary based on the model version you're referencing.

While detailed architecture files for classification aren't included in the GitHub repo as YAMLs like they are for detection, you can view the model architecture directly in the source code, specifically within the model creation scripts. This is where the transition from the shared `backbone` and `neck` to the classification-specific `head` occurs, defining the final layers for predictions.

For an in-depth look, I recommend examining the source code related to model construction, which will give you the specifics on the architecture and the number of convolutional blocks used in the head for classification. Happy coding! 🚀
